src/prefect/schedules/clocks.py

```diff
@@ -264,6 +268,11 @@ def events(self, after: datetime = None) -> Iterable[ClockEvent]:
                 microsecond=after.microsecond,
             )
         )
+
+        # Respect microseconds by rounding up
+        if after_localized.microsecond:
+            after_localized = after_localized + timedelta(seconds=1)
```

Comment:
You could compute the `timedelta` offset by directly inspecting the version of `croniter` running if you want to be targeted / precise here per your comment.

Reply:
Great idea! I'll add that in

Reply:
Do you know of a safe way to interpret version from a package that doesn't have `__version__`? croniter doesn't seem to have that 🤔

Reply:
Oh boy.  There are ways that involve metaprogramming but might not be worth it.  Here's a code snippet if you do want to take this route:

```python
import pkg_resources

[p.version for p in pkg_resources.working_set if p.project_name.lower() == "croniter"]
# returns ['0.3.25'] for me
```